forem/forem
# Introduce "extensions"

## What type of PR is this? (check all applicable)


## Description
As we are https://github.com/forem/forem/pull/16437, we need a way for admins to toggle this feature flag if they intend to keep using Listings for their Forems. We came to the conclusion that we will introduce a new route under advanced/developer tools called "Extensions" and allow users the ability to enable or disable this extension as they see fit.

You can read more about this issue [here](https://github.com/forem/forem/issues/16698).

## Related Tickets & Documents

- Closes #https://github.com/forem/forem/issues/16698

## QA Instructions, Screenshots, Recordings
<img width="1594" alt="Screenshot 2022-03-14 at 17 24 03" src="https://user-images.githubusercontent.com/2786819/158204523-c5a6a846-46a1-43e7-bb8f-a9e74477ac91.png">

1. Go to http://localhost:3000/admin/advanced/extensions.
2. Enable or disable the Listings Feature and check in the rails console that the underlying Feature Flag is getting enabled and disabled by running `FeatureFlag.enabled?(:listing_feature)`.

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?

- x] Yes
      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](gif_link)
